libminifi/src/c2/C2Client.cpp

```diff
@@ -191,23 +193,23 @@ std::shared_ptr<state::response::ResponseNode> C2Client::loadC2ResponseConfigura
 
 std::optional<state::response::NodeReporter::ReportedNode> C2Client::getMetricsNode(const std::string& metrics_class) const {
   std::lock_guard<std::mutex> guard{metrics_mutex_};
+  const auto createReportedNode = [](const std::vector<std::shared_ptr<state::response::ResponseNode>>& nodes) {
+    state::response::NodeReporter::ReportedNode reported_node;
+    reported_node.is_array = nodes[0]->isArray();
```

Comment:
A `gsl_Expects(!nodes.empty())` would make this look safer.  Or maybe pass in `name` and `is_array` as parameters?

Reply:
I think using `gsl_Expects` is a good idea here, added in bf46f4dbb1245cd6b1ba4108f08cfaf56694d0ae